pkg/awsSGSetup/aws_security_group_setup.go

```diff
@@ -0,0 +1,320 @@
+		Region:      aws.String(region),
+		Credentials: credentials.NewStaticCredentials(awsKeyId, awsSecret, ""),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to update AWS SecurityGroups: %s", err.Error())
+	}
+
+	ec2Cli := ec2.New(sess)
+
+	vpcId, err := getVPCid()
+	if err != nil {
+		return fmt.Errorf("failed to update AWS SecurityGroups: %s", err.Error())
+	}
+
+	// Get SG ids in VPC
+	// Get one with filter tag:Name with *-master-sg
+	// Get one with filter tag:Name with *-worker-sg
+	masterSg, err := getSGGroup(ec2Cli, vpcId, "*-master-sg")
+	if err != nil {
+		return fmt.Errorf("failed to get AWS SecurityGroups: %s", err.Error())
+	}
+	workerSg, err := getSGGroup(ec2Cli, vpcId, "*-worker-sg")
+	if err != nil {
+		return fmt.Errorf("failed to get AWS SecurityGroups: %s", err.Error())
+	}
+
+	// # Add rules to master and worker SG that allow incoming from master and worker for BGP, IPIP, Typha comms
+	src := []ingressSrc{
+		{
+			srcSGId:  aws.StringValue(masterSg.GroupId),
+			protocol: "tcp",
+			port:     aws.Int64(179),
+		},
+		{
+			srcSGId:  aws.StringValue(masterSg.GroupId),
+			protocol: "4",
+		},
+		{
+			srcSGId:  aws.StringValue(masterSg.GroupId),
+			protocol: "tcp",
+			port:     aws.Int64(5473),
+		},
+		{
+			srcSGId:  aws.StringValue(workerSg.GroupId),
+			protocol: "tcp",
+			port:     aws.Int64(179),
+		},
+		{
+			srcSGId:  aws.StringValue(workerSg.GroupId),
+			protocol: "4",
+		},
+		{
+			srcSGId:  aws.StringValue(workerSg.GroupId),
+			protocol: "tcp",
+			port:     aws.Int64(5473),
+		},
+	}
+	err = allowIngressToSG(ec2Cli, masterSg, src)
+	if err != nil {
+		return fmt.Errorf("failed to update master AWS SecurityGroup: %s", err.Error())
+	}
+
+	err = allowIngressToSG(ec2Cli, workerSg, src)
+	if err != nil {
+		return fmt.Errorf("failed to update worker AWS SecurityGroup: %v", err)
+	}
+
+	return nil
+}
+
+// getAWSCreds reads the aws-creds Secret that is created by an Openshift install and returns
+// the id and secret.
+func getAWSCreds(ctx context.Context, client client.Client) (id, secret string, err error) {
+	// Grab Secret kube-system aws-creds
+	//		get aws_access_key_id and aws_secret_access_key
+	creds := &v1.Secret{}
+	key := types.NamespacedName{Name: "aws-creds", Namespace: metav1.NamespaceSystem}
+
+	if err := client.Get(ctx, key, creds); err != nil {
+		return "", "", err
+	}
+	log.V(DEBUG).Info("Retrieved aws-creds")
+
+	idByte, ok := creds.Data["aws_access_key_id"]
+	if !ok {
+		return "", "", fmt.Errorf("aws-creds ConfigMap does not have key aws_access_key_id")
+	}
+	secretByte, ok := creds.Data["aws_secret_access_key"]
+	if !ok {
+		return "", "", fmt.Errorf("aws-creds ConfigMap does not have key aws_secret_access_key")
+	}
+	return string(idByte), string(secretByte), nil
+}
+
+//readMeta queries the metadata at the specified subpath on the instance.
+func readMeta(subpath string) (string, error) {
+	u, err := url.Parse("http://169.254.169.254/latest/meta-data")
```

Comment:
Did you end up investigating the `https://docs.aws.amazon.com/sdk-for-go/api/aws/ec2metadata/` and/or the `https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/` apis?

Reply:
I'm confused. I had switched to use ec2metadata but obviously lost that switch. Dang it. :angry: